src/components/search-box.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react'
+import axios from 'axios'
+import { observer } from 'mobx-react'
+
+export default @observer class SearchBox extends Component {
+  onInputChange = event => {
+    this.props.artistsStore.searchedArtist = event.target.value
+  }
+
+  render() {
+    const { artistsStore } = this.props
+
+    return (
+      <div>
+        <div className="search-box__input-container">
+          <span className="search-box__artist-input-wrapper">
+            <input
+              className="search-search-box__artist-input"
+              type="text"
+              name="artist"
+              value={artistsStore.searchedArtist}
+              onChange={this.onInputChange}
+              // Applied with each typed letter
+              onSubmit={axios
+                .get(`/api/artist/${artistsStore.searchedArtist}`)
```

Comment:
Wydaje mi się, że nie potrzebujesz `searchedArtist` zapisywać w store. Chyba nigdzie indziej z tego nie korzystasz. Lepiej zrobić lokalną zmienna w klasie.

Reply:
Obecnie nie, ale wole miec w storze jak bede dodawac nowa funkcjonalnosc + pomaga mi w nauce storow hehe

Reply:
Hmm jeżeli nie masz jakiegoś konkretnego ficzera, którego będziesz chciał dodać, który będzie wymagał tego jako globalnej zmiennej to wydaje mi się, że lepiej jednak to zamienić. Zawsze łatwiej jest coś dodać niż usunąć i robienie rzeczy na zapas jest często niepotrzebne. A stan globalny to raczej coś czego lepiej unikać :)

Reply:
hmm okej, chcialem dawac rzeczy do store, bo sie ucze mobx w sumie. :D  Zmienilem to troche, zerknij teraz.